### DISCUSSION TITLE
How are the examples tested

### DISCUSSION THREAD

**Discussion by wigging** [mwaskom/seaborn#3096] [Q&A] [answered]
In the seaborn repository there's an examples directory at the top of the project. How are these example files tested with pytest?

**Comment by mwaskom** [comment, marked as answer]
They're not tested with pytest but they are the source of the [example gallery](https://seaborn.pydata.org/examples/) in the docs and get executed as part of the doc build in CI.

**Reply by wigging** [reply]
So if an example doesn't work then the docs fail to build?

**Reply by mwaskom** [reply]
Yes. Why?

**Reply by wigging** [reply]
Ok, so the examples are basically tested by whether the docs successfully build or fail.

**Reply by mwaskom** [reply]
Right. Sorry, is there a point to this line of questioning?

**Reply by wigging** [reply]
There are a lot of Python packages developed on GitHub. The repositories for these projects usually contain an examples directory of scripts or Jupyter notebooks that demonstrate how to use the package. I'm curious about how these projects ensure that the examples continue to work as new versions of the package are released.
